src/index.ts

```diff
@@ -137,17 +138,27 @@ export const createRouter = <
       [JSON.stringify(routeNames)],
     );
 
-    const routes = useSyncExternalStore(subscribe, () => {
-      const routes = matchAll(getCurrentLocation(), matchers);
+    const { routes } = useSyncExternalStoreWithSelector(
+      subscribe,
+      () => {
+        const routes = matchAll(getCurrentLocation(), matchers);
 
-      if (orderBy === "asc") {
-        routes.reverse();
-      }
+        if (orderBy === "asc") {
+          routes.reverse();
+        }
 
-      return JSON.stringify(routes);
-    });
+        return {
+          key: JSON.stringify(routes),
+          routes,
+        };
+      },
+      undefined,
+      identity,
+      areItemKeysEqual,
+    );
 
-    return JSON.parse(routes);
+    // @ts-expect-error
+    return routes;
```

Comment:
In the end, I feel like we can remove that entire function (as we found another more efficient way to handle breadcrumbs patterns internally)

Reply:
Good for me. I never really liked it.